srcs/builtin/_environ_utils.c

```diff
@@ -0,0 +1,134 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   _environ_utils.c                                   :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: kitsuki <kitsuki@student.42tokyo.jp>       +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/05/22 02:08:46 by kitsuki           #+#    #+#             */
+/*   Updated: 2023/05/22 02:08:46 by kitsuki          ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "ft_string/ft_string.h"
+#include "_environ.h"
+#include "_util_commands.h"
+
+bool	set_environs(char **src)
+{
+	char	**tmp;
+	char	**dst;
+
+	dst = NULL;
+	if (src != NULL)
+	{
+		dst = (char **)malloc((get_strslen(src) + 1) * sizeof (char *));
+		if (dst == NULL)
+			return (false);
+		tmp = dst;
+		while (*src != NULL)
+		{
+			*tmp = ft_strdup(*src);
+			if (*tmp == NULL)
+				return (free_all(&dst));
+			tmp++;
+			src++;
+		}
+		*tmp = NULL;
+	}
+	free_all(*get_environs());
+	*get_environs() = dst;
+	return (true);
+}
+
+char	**search_environ(const char *name)
+{
+	char	**env;
+	size_t	size;
+
+	env = *get_environs();
+	if (env == NULL || name == NULL)
+		return (NULL);
+	while (*env != NULL)
+	{
+		size = ft_strlen(name);
```

Comment:
`name`はconstなので、このstrlenは外に出しちゃいましょう

```suggestion
	size = ft_strlen(name);
	while (*env != NULL)
	{
```